convert2rhel/main.py

```diff
@@ -23,6 +23,7 @@
 from convert2rhel import logger as logger_module
 from convert2rhel import pkghandler, pkgmanager, redhatrelease, repo, subscription, systeminfo, toolopts, utils
 from convert2rhel.actions import level_for_raw_action_data, report
+from convert2rhel.actions.pre_ponr_changes.backup_system import BackupPackageFiles
```

Comment:
That would mean that both the action and the other module have shared data... This must be controlled very well on the other module side to prevent errors if the BackupPackageFiles does not run. 

I think that the BackupPackageFiles can be a action that only writes to that shared data structure and the other module (probably pkghandler (or backup?)) does the rollback if necessary... 

Does that introduce more risks than it is intended? I don't think see another solution other than that to use with the action framework... Well, maybe another idea would be a way to tell the action framework like: "Run this action in case of failure", kinda of a new stage that is only triggered in case of failure, with that in mind, we would have two actions: 
1. BackupPackageFiles - to backup the packages and have it in a shared data structure
2. RollbackPackageFiles - to restore what the BackupPackageFiles saved

But that would be too much work for now, right? Probably the first idea sounds more straightforward to do.

Reply:
I also see that this PR has a dependency on https://github.com/oamg/convert2rhel/pull/935, so my comment may not be appropriate at all.